swift/extractor/infra/SwiftDispatcher.h

```diff
@@ -215,6 +215,17 @@ class SwiftDispatcher {
   // Emits a Location TRAP entry and attaches it to a `Locatable` trap label
   template <typename Locatable>
   void attachLocation(Locatable* locatable, TrapLabel<LocatableTag> locatableLabel) {
+    // Swift 5.7.1 causing a crash when the Swift compiler uses .back() of and empty
+    // getPatternList() when getting the source location.
+    // So far this only observed with the entities coming from precompiled modules, so they
+    // should not have locations anyway.
+    if constexpr (std::is_base_of<swift::Decl, Locatable>::value) {
+      if (auto* decl = llvm::dyn_cast<swift::PatternBindingDecl>(locatable)) {
+        if (decl->getPatternList().empty()) {
+          return;
+        }
+      }
+    }
```

Comment:
I don't think you need a `dyn_cast` here because `Locatable` is already `const swift::PatternBindingDecl` here. And also, rather than not giving any location, we can use the (existing) start location as and location as well.
So maybe the following works as well:
```suggestion
    if constexpr (std::is_same_v<const swift::PatternBindingDecl, Locatable>) {
      if (locatable->getPatternList().empty()) {
        attachLocation(locatable->getStartLoc(), locatableLabel);
        return;
      }
    }
```
Apart from that, this is a bit ewww :sweat_smile: 

Shouldn't we maybe consider patching this on `PatternBindingDecl::getSourceRange()` before prebuilding? We already have patches on the swift frontend library, although for the moment it's just for the build system, right? If we go for that we could also move there the patch I already introduced for headers.

Reply:
Yeah, patching the source code makes more sense, agreed, I'll clean it up when we resolve all the rest of the issues if it doesn't cause too much friction.

The suggested change won't work unfortunately as the crash is coming right from `Decl::getStart/EndLoc` which  subsequently calls `getSourceRange`.

Reply:
ah, you are right, this code is called in `fetchLabel` that might be called with a less specific type than the translation actually uses. If we moved the location attachment to `createEntry` instead, we would always get the exact specific type and we would be able to do this kind of tweaks to locations on a per-type basis. But we can leave that for another PR